packages/ui/src/components/Button/SButton.vue

```diff
@@ -0,0 +1,248 @@
+<script lang="ts">
+export default defineComponent({ name: 'SButton' })
+</script>
+
+<script setup lang="ts">
+import { computed, ComputedRef } from 'vue'
+import { SSpinner } from '@/lib'
+
+const BUTTON_TYPE_VALUES = ['primary', 'secondary', 'outline', 'action'] as const
+const BUTTON_SIZE_VALUES = ['mini', 'small', 'medium', 'big'] as const
+const BUTTON_ICON_POSITION_VALUES = ['left', 'right'] as const
+
+type ButtonType = typeof BUTTON_TYPE_VALUES[number]
+type ButtonSize = typeof BUTTON_SIZE_VALUES[number]
+type ButtonIconPosition = typeof BUTTON_ICON_POSITION_VALUES[number]
+
+const SPINNER_SIZE: Record<ButtonSize, string> = {
+  mini: '12',
+  small: '18',
+  medium: '24',
+  big: '24',
+}
+
+const SPINNER_WIDTH: Record<ButtonSize, string> = {
+  mini: '3',
+  small: '4',
+  medium: '4',
+  big: '4',
+}
+
+function usePropTypeFilter<T>(value: T, validValues: readonly T[], defaultValue: T): ComputedRef<T> {
+  return computed(() => {
+    if (validValues.includes(value)) {
+      return value
+    }
+
+    return defaultValue
+  })
+}
+
+const props = withDefaults(
+  defineProps<{
+    type?: ButtonType,
+    size?: ButtonSize,
+    nativeType?: 'button' | 'reset' | 'submit',
+    icon?: string,
+    iconPosition?: ButtonIconPosition,
+    rounded?: boolean,
+    disabled?: boolean,
+    loading?: boolean,
+    uppercase?: boolean,
+  }>(),
+  {
+    type: 'secondary',
+    size: 'medium',
+    nativeType: 'button',
+    icon: '',
+    iconPosition: 'left',
+    rounded: false,
+    disabled: false,
+    loading: false,
+    uppercase: false,
+  },
+)
+
+const definitelyType = usePropTypeFilter(props.type, BUTTON_TYPE_VALUES, 'primary')
+const definitelySize = usePropTypeFilter(props.size, BUTTON_SIZE_VALUES, 'medium')
```

Comment:
Hmmm, isn't it better to use native Vue prop validations? In this case Vue will print warnings, but with current implementation nothing is warned.